### DISCUSSION TITLE
`pv_power` is always 0 (how to change to plant_pv_power?)

### DISCUSSION THREAD

**Discussion by lhjt** [TypQxQ/Sigenergy-Home-Assistant-Integration#43] [Q&A] [answered]
Hi,

Thanks for your work on this integration. Everything works fine, except the `sensor.sigen_pv_power` always returns 0, even though we have installed solar panels.

I have noticed that the `sensor.sigen_plant_pv_power` shows the correct expected values, but the dashboard seems to rely on the `sigen_pv_power` instead. Is it possible to easily change everything to rely on the `sigen_plant_pv_power` or how should I approach this?

Thanks!

**Comment by TypQxQ** [comment]
It works fine on my system. Could it be your setup is configured in an unusual way?

![image](https://github.com/user-attachments/assets/a9482500-8d5d-46dc-bb6c-03471b10a682)

You could just search and replace instances of the one sensor with the other in your files.

**Reply by lhjt** [reply]
Thanks for the advice. I will have a look at the search and replace method to see if it fixes my problem, but I don't think I have an unusual setup? At least nothing custom was set up.

So you specifically recommend search and replacing `sigen_pv_power` with `sigen_pv_plant_power`?

**Reply by lhjt** [reply]
![20241221T055007-Arc-Developer tools – Home Assistant](https://github.com/user-attachments/assets/112fb989-f19a-4db6-aeaa-37dac0eea6b1)

@TypQxQ is this what it's expected to look like?

**Reply by TypQxQ** [reply]
No, you need to change the code. It works for me and seemingly everyone else with the standard settings. I think your installer did something strange. 

If you want to change the code, it's the files you've downloaded from GitHub.

**Reply by lhjt** [reply, marked as answer]
Thanks for your help so far. I spent the weekend debugging and managed to get it fixed - it appears that this is certainly a "non-standard" install.

In my case, my PV panels are connected to an existing external inverter, which is then connected to the Sigen system. Therefore, each of the PV values (PV1-4) appears to show 0 as I don't believe Sigen has access to the panel-level data?

The solution was to replace all references to `sigen_pv_power` with `sigen_plant_pv_power` in the config. The daily totals for power now work as expected; however, `sigen_pv_power` and all `PV{1,4}` remain 0.

**Reply by TypQxQ** [reply]
That explains it. I'm happy that it worked out in the end.

**Reply by StroppyOz** [reply]
I have this same issue and from what the OP said, I have a similar setup in that I am using an external inverter. Moving forward may I suggest the wiki installation instructions be updated to reflect what needs to be done for those who are using external inverters? I'm happy to contribute to these changes.

**Reply by StroppyOz** [reply]
@lhjt when you say you replaced all references in the config, do you mean on the dashboard or the modbus_signenergy.yaml file (or both)?

**Reply by lhjt** [reply]
Hi @StroppyOz - I at least modified the `modbus_sigenergy.yaml` file. It's been a while so I don't remember if I modified the dashboard after that, but it may be very possible (sorry this probably doesn't clear it up that much 😓)

**Comment by EnterTheMaelstrom** [comment]
Hi, do you have an energy meter in your installation? 
Maybe this is not installed correctly or the data cable is damaged. That might explain why you get information from the plant, but not the general pv power.

When you open the app, under "device" there are (at least) two devices under "sigen device". One should be the PV modules, one a power/energy meter and the maybe gateway and charger. Does the power/energy meter show values ?

**Reply by lhjt** [reply]
Thanks for helping out - in my case the values were appearing in the Sigen app but as I have outlined in my selected answer above, I have an external inverter that is coupled to the Sigen system, rather than the PV system being connected directly to the Sigen inverter (without an intermediary). This means I just have to rely on the plant values.

**Comment by goatie999** [comment]
Slightly off topic but has anyone ever worked out the actually definition difference between **PV Power** and **Plant PV Power**?  The readings always close but never the same. 

![IMG_7347](https://github.com/user-attachments/assets/d90d1212-65bd-427b-88e0-766b66938c11)

**Reply by TypQxQ** [reply]
I think they're not the same because they're not read at exactly the same millisecond.
As I understand it Plant can consist of one or several devices. If only one Sigenergy device then it will be the same but multiple Sigenergy devices can be connected together to form a plant. The first device has id 0 and the other devices can have other ids.

Maybe the OPs device has another id than 0 set up?

**Reply by Janneck79** [reply]
It looks like plant power is the total power minus the power the inverter uses

**Reply by azebro** [reply]
Exactly that is my understanding. PV pwer is what is generated by the panels and plant pv power is net available. For me difference is usually quite substantial though (150-200W) and that is a lot for inverter internal usage, inefficiency, etc.
I am wondering is anybody managed to measure the actual power draw by the Sigenstore and gateway, as I hope I am not paying almost 4kWh a day just to run the system.

**Reply by Janneck79** [reply]
Sigen inverter power use is the sensor at least for the inverter. 10kW inverter with 98% efficiency is around 200W poweruse of the inverter. The problem with the efficiency in the data sheets is that it is shown for the best case    and the inverter uses as soon solar is producing more than 100W! I had the system installed in the beginning of Dezember and narly the half solar production until now was used by the inverter.

**Reply by ZB-42** [reply]
I've been tracking the "self-consumption" of my system for a while now. There seem to be two components, one is a reduction in the available power from the PV array during the day and the other is is a reduction in the available power from the battery overnight. In my system the losses vary but on average they are around 0.3kW and 0.1kW respectively. I have set up a utility meter to track this and it  translates into losses of around 5kWh per day.  This is in a system that has a solar production of around 60kWh and therefore represents about 8%.
